File: src/JpaMain.java
import a.b.c.Sample;
import a.b.c.Tab2;

import javax.persistence.EntityManager;
import javax.persistence.EntityManagerFactory;
import javax.persistence.Persistence;
import javax.persistence.Query;

public class JpaMain {

    public static void main(String[] args) {
        // Open a database connection
        // (create a new database if it doesn't exist yet):
        EntityManagerFactory factory = Persistence.createEntityManagerFactory("NewPersistenceUnit");

        EntityManager entitymanager = factory.createEntityManager();
        entitymanager.getTransaction().begin();
// Find the number of objects in the database:
        Query count = entitymanager.createQuery("SELECT count(s) FROM Sample s");
        final Long result = (Long) count.getSingleResult();
        int rez = Math.toIntExact(result + 1);
        System.out.println(result);
        Sample sample = new Sample();
        sample.setId(rez);
        sample.setSample("sample" + rez);
        sample.setColor("red");
        sample.setVersion(1);
        entitymanager.persist(sample);
        entitymanager.getTransaction().commit();


        Query namedQuery2 = entitymanager.createNamedQuery("myQuery2");
        System.out.println("**********************************");
        System.out.println("executing the named query from Sample");
        System.out.println("**********************************");
        for (Object o : namedQuery2.getResultList()) {
            Sample s = (Sample) o;
            System.out.println(s.toString());
        }
        Query namedQuery1 = entitymanager.createNamedQuery("myQuery1");
        System.out.println("**********************************");
        System.out.println("executing the named query from Tab2");
        System.out.println("**********************************");
        for (Object o : namedQuery1.getResultList()) {
            Tab2 s = (Tab2) o;
            System.out.println(s.toString());
        }
        Query nativeQuery1 = entitymanager.createNamedQuery("nativeQueryForSample");
        System.out.println("**********************************");
        System.out.println("executing the native query from Sample");
        System.out.println("**********************************");
        for (Object o : nativeQuery1.getResultList()) {
            Sample s = (Sample) o;
            System.out.println(s.toString());
        }
        // Close the database connection:
        entitymanager.close();
        factory.close();
    }

}

File: src/JpaMain.kt
import a.b.c.Sample
import a.b.c.Tab2
import javax.persistence.Persistence

object JpaMain {

    @JvmStatic
    fun main(args: Array<String>) {
        // Open a database connection
        // (create a new database if it doesn't exist yet):
        val factory = Persistence.createEntityManagerFactory("NewPersistenceUnit")

        val entitymanager = factory.createEntityManager()
        entitymanager.transaction.begin()
        // Find the number of objects in the database:
        val count = entitymanager.createQuery("SELECT count(s) FROM sample s")
        val result = count.singleResult as Long
        val rez = Math.toIntExact(result + 1)
        println(result)


        /*  val sample = Sample()
          sample.id = rez
          sample.sample = "sample$rez"
          sample.color = "red"
          sample.version = 1
          entitymanager.persist(sample)
          entitymanager.transaction.commit()*/


        val namedQuery2 = entitymanager.createNamedQuery("namedQueryForSample")
        println("**********************************")
        println("executing the named query 'namedQueryForSample'")
        println("**********************************")
        for (o in namedQuery2.resultList) {
            val s = o as Sample
            println(s.toString())
        }

        val namedQuery1 = entitymanager.createNamedQuery("namedQueryForTab2")
        println("**********************************")
        println("executing the named query 'namedQueryForTab2'")
        println("**********************************")
        for (o in namedQuery1.resultList) {
            val t = o as Tab2
            println(t.toString())
        }
        val nativeNamedQuery1 = entitymanager.createNamedQuery("nativeQueryForSample")
        println("**********************************")
        println("executing the native query 'nativeQueryForSample'")
        println("**********************************")
        for (o in nativeNamedQuery1.resultList) {
            val s = o as Sample
            println(s.toString())
        }

        // Close the database connection:
        entitymanager.close()
        factory.close()
    }

}
